### DISCUSSION TITLE
Embarking on the journey to run BRAT...

### DISCUSSION THREAD

**Discussion by ecopony** [Riverscapes/riverscapes-tools#693] [BRAT]
Hi all,
I just recently came across your project. I'm interested in running BRAT against some Oregon hydrologic units potentially to use for a master's thesis in GIS.

I see this in the README:

> It is strongly advised that you continue to use pyBRAT 3 until this new version of BRAT is verified and published as an official version of the BRAT model.

And I see these docs aren't fleshed out quite yet: https://tools.riverscapes.net/brat/Support/Advanced_Users/running_brat.html

I would really like to use the new version if possible and not be tied to ArcGIS -- I just wanted to check in and see if that warning was still applicable.

Thanks!
Ed

**Comment by ecopony** [comment]
Just checking in on this thread. I took the sqlbrat route. 

By picking through the docs and source code, I was able to get through a Brat run on macOS.

I'm curious if the project team is interested in contributions to the documentation.

Thanks!

**Reply by philipbaileynar** [reply]
Absolutely please make pull requests onto the docs branch.

**Reply by GeoNicole** [reply]
Hey! We're looking at running this in BC (Canada) as well - working with a student who doesn't have access to ArcMap/Py2.x for pyBrat so we're trying to run sqlBrat in windows/Pro/Python 3.7 as well - Would you be able to share your details with us?

**Reply by ecopony** [reply]
Hi @GeoNicole -- I haven't gone through my notes yet to produce something worthy of a pull request. My notes are also specific to macOS.

There are some Windows-specific docs that I found helpful here: https://tools.riverscapes.net/brat/Support/Advanced_Users/installation.html

Once I was through the bulk of the dependencies, the most helpful thing was referencing some of the automation scripts in the project for clues on what I would need to do locally to get all the way through a BRAT run. This script has the whole flow outlined and was the most useful reference I found: https://github.com/Riverscapes/riverscapes-tools/blob/master/scripts/automation/FargateAllTools.sh

The main thing to know is that you can't just 'run BRAT.' You need to run a sequence of tools, each producing an output that the next tool requires. You can see that sequence in that automation script (rscontext -> channel -> taudem -> vbet -> brat build -> brat run). I ran into a handful of issues as I went through them all for the first time that were mostly dependency issues with my environment.

I have designs to write a 'Getting Started for Developers' guide with a bunch of info, I just haven't gotten around to it. I attached my raw notes -- hopefully they might be of some use.

[GEOS-BRAT - Running BRAT-210623-010659.pdf](https://github.com/Riverscapes/riverscapes-tools/files/11811100/GEOS-BRAT.-.Running.BRAT-210623-010659.pdf)

**Comment by GeoNicole** [comment]
Awesome! Thank you! Using the installation guide for reference we've got what we think will work for a Python environment set up already looking at the dependencies/requirements but it hasn't been tested yet. The main sticking part is figuring out the brat build and brat run and how to feed things into that when we're not using the 'standard' workflow. Where we're not using US data, all the data extraction, prep, pre-processing, flow calcs, etc has already been done outside of BRAT.
That shell script and the others with it looks awesome though, thank you for pointing me there. I'll sort through your notes and those scripts and with that I'm hoping we can cobble something together as well. If we're able to get things working I'll post our rough notes here too!


________________________________
